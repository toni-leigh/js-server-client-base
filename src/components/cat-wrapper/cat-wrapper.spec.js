import React from 'react';
import { shallow } from 'enzyme';
import { createStore } from 'redux';
import CatWrapperConnected, { CatWrapper } from './cat-wrapper';
import rootReducer from '../../reducers/reducers';

const store = createStore(rootReducer);

describe('<CatWrapperConnected />', () => {
  let catWrapperConnected;

  beforeAll(() => {
    catWrapperConnected = shallow(
      <CatWrapperConnected store={ store }/>
    );
  });

  test('render occurs through connection', () => {
    expect(catWrapperConnected).toMatchSnapshot();
  });
});

describe('<CatWrapper />', () => {
  let catWrapper;

  beforeAll(() => {
    catWrapper = shallow(
      <CatWrapper
        getCat={ jest.fn() }
        imageSrc='http://test.jpg'
      />
    );
  });

  test('basic dumb component render', () => {
    expect(catWrapper).toMatchSnapshot();
  });
});
