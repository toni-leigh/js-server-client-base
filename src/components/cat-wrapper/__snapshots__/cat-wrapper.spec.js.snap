// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatWrapper /> basic dumb component render 1`] = `
<div
  className="crardx-cat-wrapper"
>
  <CatImage
    imageSrc="http://test.jpg"
  />
  <CatButton
    getCat={[MockFunction]}
  />
</div>
`;

exports[`<CatWrapperConnected /> render occurs through connection 1`] = `
<withImmutablePropsToJS(CatWrapper)
  getCat={[Function]}
  imageSrc=""
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`mapDispatchToProps processes the actions correctly 1`] = `
Object {
  "getCat": [Function],
}
`;

exports[`mapStateToProps processes the state correctly 1`] = `
Object {
  "imageSrc": "http://test.jpg",
}
`;
